sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeCreator.scala

```diff
@@ -441,10 +452,15 @@ case class StringToMap(text: Expression, pairDelim: Expression, keyValueDelim: E
         UTF8String[] $keyArray = new UTF8String[$tempArray.length];
         UTF8String[] $valueArray = new UTF8String[$tempArray.length];
 
-        for (int $i = 0; $i < $tempArray.length; $i ++) {
+        for (int $i = 0; $i < $tempArray.length; $i++) {
           UTF8String[] $keyValue = ($tempArray[$i]).split($keyValueDelim, 2);
           $keyArray[$i] = $keyValue[0];
-          $valueArray[$i] = $keyValue[1];
+          if ($keyValue.length < 2) {
+            $valueArray[$i] = null;
+          }
+          else {
+            $valueArray[$i] = $keyValue[1];
+          }
```

Comment:
I think that syntax is allowed in java, and it's much more readable than tertiary if notation.